clar2wasm/src/wasm_generator.rs

```diff
@@ -402,12 +413,15 @@ impl WasmGenerator {
                 memory: memory.id(),
                 location: walrus::ActiveDataLocation::Absolute(offset),
             }),
-            data.clone(),
+            data,
         );
-        self.literal_memory_end += data.len() as u32;
+        self.literal_memory_end += len;
 
-        // Save the offset in the literal memory for this string
-        self.literal_memory_offet.insert(s.to_string(), offset);
+        // Only save literals for ASCII
+        if let CharType::ASCII(_) = s {
+            // Save the offset in the literal memory for this string
+            self.literal_memory_offet.insert(s.to_string(), offset);
+        }
```

Comment:
I'm not sure if/how utf8 strings should share space with ascii strings in this hashmap. Is there a reliable way to prevent key collisions?

Reply:
First, how did I never notice that typo before in the multiple places I used it?!

I guess we should add something to the key to indicate which it is. The literals used for other non-clarity strings (e.g. function names) should use the ascii keys.

Reply:
We could also make the key an `Enum` and leave the string as-is?

Reply:
That sounds cleaner, something like:
```rs
pub enum LiteralMemoryEntry {
    Ascii(String),
    Utf8(String),
}
```
?

Reply:
Done